crates/astria-sequencer/src/action_handler/impls/update_params.rs

```diff
@@ -0,0 +1,117 @@
+use astria_core::protocol::transaction::v1::action::UpdateParams;
+use astria_eyre::eyre::{
+    self,
+    ensure,
+    Context,
+};
+
+use crate::{
+    action_handler::ActionHandler,
+    authority::StateReadExt as _,
+    connect::market_map::state_ext::StateWriteExt as _,
+    transaction::StateReadExt as _,
+};
+
+#[async_trait::async_trait]
+impl ActionHandler for UpdateParams {
+    async fn check_stateless(&self) -> eyre::Result<()> {
+        Ok(())
+    }
+
+    async fn check_and_execute<S: cnidarium::StateWrite>(&self, mut state: S) -> eyre::Result<()> {
+        let from = &state
+            .get_transaction_context()
+            .expect("transaction source must be present in state when executing an action")
+            .address_bytes();
+        let sudo_address = state
+            .get_sudo_address()
+            .await
+            .wrap_err("failed to get sudo address from state")?;
+        ensure!(sudo_address == *from, "signer is not the sudo key");
```

Comment:
i think this should be the market map admin, actually

also, i think the name of this action is too unclear/generic, we should rename it to `UpdateMarketMapParams`

Reply:
interesting, seems okay to leave this then!